pkg/controller/tailoredprofile/tailoredprofile_controller.go

```diff
@@ -254,39 +343,64 @@ func (r *ReconcileTailoredProfile) Reconcile(ctx context.Context, request reconc
 
 	}
 
-	rules, ruleErr := r.getRulesFromSelections(instance, pb)
-	if ruleErr != nil && !common.IsRetriable(ruleErr) {
-		// Surface the error.
-		suerr := r.handleTailoredProfileStatusError(instance, ruleErr)
-		return reconcile.Result{}, suerr
-	} else if ruleErr != nil {
-		return reconcile.Result{}, ruleErr
-	}
+	// we should do the following only if the tailored profile does not have CustomRules
+	if len(customRules) <= 0 {
```

Comment:
A future improvement we can make here is to use a factory (similar to what we do with scan type handlers).

The TailoredProfile controller (reconcile loop) validates the profile, and grabs a handler from a factory based on the rules in the profile. Then the reconcile loop just calls the same functions regardless of the handler.

That can be a good way to cut down on all the if/else conditionals we have here.